apps/ledger-live-mobile/src/newArch/features/Market/components/MarketRowItem/index.tsx

```diff
@@ -0,0 +1,114 @@
+import { useLocale } from "~/context/Locale";
+import DeltaVariation from "LLM/features/Market/components/DeltaVariation";
+import { counterValueFormatter } from "LLM/features/Market/utils";
+
+type Props = {
+  index: number;
+  item: CurrencyData;
+  counterCurrency?: string;
+};
+
+function MarketRowItem({ item, index, counterCurrency }: Props) {
+  const { locale } = useLocale();
+  const { t } = useTranslation();
+  const {
+    internalCurrency,
+    image,
+    name,
+    marketcap,
+    marketcapRank,
+    price,
+    priceChangePercentage,
+    ticker,
+  } = item;
+
+  let loc = locale;
+  // TEMPORARY : quick win to transform arabic to english
+  if (locale === "ar") {
+    loc = "en";
+  }
+  return (
+    <Flex
+      height={72}
+      flexDirection="row"
+      justifyContent="flex-start"
+      alignItems="center"
+      py="16px"
+      key={index}
+    >
+      {internalCurrency ? (
+        <CircleCurrencyIcon
+          size={32}
+          currency={internalCurrency}
+          color={undefined}
+          sizeRatio={0.9}
+        />
+      ) : (
+        image && (
+          <StyledIconContainer>
+            <Image source={{ uri: image }} style={{ width: 30, height: 30 }} resizeMode="contain" />
+          </StyledIconContainer>
+        )
+      )}
+      <Flex mx="4" flexDirection="column" justifyContent="center" alignItems="flex-start" flex={1}>
+        <Text variant="large" fontWeight="semiBold" numberOfLines={1}>
+          {`${name} (${ticker.toLocaleUpperCase()})`}
+        </Text>
+        <Flex flexDirection="row" alignItems="center">
+          <Text
+            variant="small"
+            bg="neutral.c40"
+            height="15px"
+            lineHeight="15px"
+            px="4px"
+            mr="3"
+            borderRadius={4}
+            overflow="hidden"
+            fontWeight="semiBold"
+          >
+            {marketcapRank || "-"}
+          </Text>
+          <Text variant="body" color="neutral.c70" fontWeight="semiBold">
+            {marketcap && marketcap > 0
+              ? counterValueFormatter({
+                  value: marketcap,
+                  shorten: true,
+                  currency: counterCurrency,
+                  locale: loc,
+                  t,
+                })
+              : "-"}
+          </Text>
+        </Flex>
+      </Flex>
+      <Flex flexDirection="column" justifyContent="center" alignItems="flex-end">
+        <Text variant="large" fontWeight="semiBold">
+          {counterValueFormatter({
+            value: price,
+            currency: counterCurrency,
+            locale: loc,
+            t,
+          })}
+        </Text>
+
+        {priceChangePercentage !== null && !isNaN(priceChangePercentage) ? (
+          <DeltaVariation percent value={priceChangePercentage} />
+        ) : (
+          <Text variant="body" color="neutral.c70">
+            {" "}
+            -
+          </Text>
+        )}
+      </Flex>
+    </Flex>
```

Comment:
[ASK] I know, you didn't refactor components but here some logic should be directly inside the component (countervaluesformatter, isNaN, etc.), you didn't not change it right ?

Reply:
Yes I did not change it and imo we can keep this logic in the view as it's related to the display (we are between two worlds), it depends on the local + the function format the number for what we want to display here. We could do it in a container too, but as we don't have one here, looks like overkill to create one just for this.